src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -209,17 +208,9 @@ protected void startup() throws InterruptedException {
                         logger.atInfo().kv(DEPLOYMENT_ID_LOG_KEY_NAME, currentDeploymentTaskMetadata.getDeploymentId())
                                 .log("The current deployment cannot be cancelled");
                     }
-                } else if (deployment.getDeploymentType().equals(DeploymentType.SHADOW)) {
+                } else if (DeploymentType.SHADOW.equals(deployment.getDeploymentType())) {
                     // The deployment type is shadow
-                    if (deployment.getId()
-                            .equals(Coerce.toString(config.lookup(LAST_SUCCESSFUL_SHADOW_DEPLOYMENT_ID_TOPIC)))) {
-                        // On device start up, Shadow listener will fetch the shadow and schedule a shadow deployment
-                        // Discard the deployment if Kernel starts up from a tlog file
-                        // and has already processed deployment
-                        logger.atInfo().kv(DEPLOYMENT_ID_LOG_KEY_NAME, deployment.getId())
-                                .log("Skip the already processed shadow deployment");
-                        deploymentQueue.remove();
```

Comment:
why aren't we removing from the queue anymore?

Reply:
This check is basically moved here and we won't queue the duplicated deployment any more: https://github.com/aws-greengrass/aws-greengrass-nucleus/pull/827/files#diff-59c7fe23a5126f4c98b9b2baf83faa49534d72dbff905244da12031b7c523392R272-R276
This check was not covering cases other than last deployment succeeded. The new logic will cover those as well.